### PR TITLE
Add docs for MIG on YARN [skip ci]

### PR DESCRIPTION
**Pull request by tgravescs**
Currently relies on https://github.com/NVIDIA/spark-rapids-examples/pull/51 for the link to the 3.1.2 thru 3.3.0 documentation link.

skip ci since just docs.

fixes https://github.com/NVIDIA/spark-rapids/issues/4023

### CODE REVIEW

**Comment by tgravescs** [comment]
build

**Review by jlowe — APPROVED** [review]

File: docs/get-started/getting-started-on-prem.md [hunk 1]
```diff
@@ -296,6 +299,32 @@ $SPARK_HOME/bin/spark-shell \
...
+```
+
+This tells the RAPIDS Accelerator for Spark plugin to look for the Spark GPU resource assigned to it using the name `nvidia/miggpu`.
+If you are using the Spark config `spark.yarn.resourceGpuDeviceName` and using the normal `gpu` Spark resource name, this is not required.
+
+#### YARN version 3.1.2 until 3.3.0
+If you are using YARN version from 3.1.2 up until 3.3.0, it requires making modifications to YARN and deploying
+a version that adds support for MIG to the built-in YARN GPU resource plugin.
+
+See [NVIDIA Support for GPU for YARN with MIG support for YARN 3.1.2 until YARN 3.3.0](https://github.com/NVIDIA/spark-rapids-examples/blob/branch-21.10/hadoop/resource-types/gpu-mig/README.md)
```
Comment (jlowe) [inline docs/get-started/getting-started-on-prem.md:325]: This link is broken, I think it needs to be:
```suggestion
See [NVIDIA Support for GPU for YARN with MIG support for YARN 3.1.2 until YARN 3.3.0](https://github.com/NVIDIA/spark-rapids-examples/blob/branch-21.10/hadoop/device-plugins/gpu-mig/README.md)
```

**Review by jlowe — CHANGES_REQUESTED** [review]

File: docs/get-started/getting-started-on-prem.md (hunk 1 above)
Comment (tgravescs) [reply docs/get-started/getting-started-on-prem.md:325]: actually that is being added under the PR referenced in the description

Comment (jlowe) [reply docs/get-started/getting-started-on-prem.md:325]: Ah sorry. missed that the other docs were still in flight.

**Review by jlowe — APPROVED** [review]

File: docs/get-started/getting-started-on-prem.md [hunk 2]
```diff
@@ -296,6 +299,32 @@ $SPARK_HOME/bin/spark-shell \
...
+#### YARN version 3.3.0+
+YARN version 3.3.0 and newer support a pluggable device framework which allows adding support for MIG devices via a plugin.
+See [NVIDIA GPU Plugin for YARN with MIG support](https://github.com/NVIDIA/spark-rapids-examples/blob/branch-21.10/hadoop/device-plugins/gpu-mig/README.md).
+If you are using that plugin with a Spark version older than 3.2.1 and/or specifying the resource as `nvidia/miggpu` you will also need to specify the config:
+
+```shell
+--conf spark.rapids.gpu.resourceName=nvidia/miggpu
+```
+
+This tells the RAPIDS Accelerator for Spark plugin to look for the Spark GPU resource assigned to it using the name `nvidia/miggpu`.
```
Comment (gerashegalov) [inline docs/get-started/getting-started-on-prem.md:318]: consistency with other occurrences  in the docs nit:
```suggestion
This tells the RAPIDS Accelerator for Apache Spark plugin to look for the Spark GPU resource assigned to it using the name `nvidia/miggpu`.
```

**Review by gerashegalov — DISMISSED** [review]

**Comment by tgravescs** [comment]
build

**Review by gerashegalov — APPROVED** [review]

File: docs/get-started/getting-started-on-prem.md [hunk 3]
```diff
@@ -296,6 +299,32 @@ $SPARK_HOME/bin/spark-shell \
...
+Using MIG enabled GPUs on YARN requires enabling YARN GPU scheduling, cgroups, and using NVIDIA Docker runtime v2.
+The way to set this up depends on the version of YARN and the version of Spark.
+It is important to note that CUDA 11 only supports enumeration of a single MIG instance. This means that using
+any MIG device on YARN means only 1 GPU per container is allowed. See the limitations section in the documentation
+referred to below for the specific YARN version you are using.
+
+#### YARN version 3.3.0+
+YARN version 3.3.0 and newer support a pluggable device framework which allows adding support for MIG devices via a plugin.
+See [NVIDIA GPU Plugin for YARN with MIG support](https://github.com/NVIDIA/spark-rapids-examples/blob/branch-21.10/hadoop/device-plugins/gpu-mig/README.md).
+If you are using that plugin with a Spark version older than 3.2.1 and/or specifying the resource as `nvidia/miggpu` you will also need to specify the config:
```
Comment (sameerz) [inline docs/get-started/getting-started-on-prem.md:312]: Minor: try to keep line lengths <= 100 charactors

File: docs/get-started/getting-started-on-prem.md [hunk 4]
```diff
@@ -296,6 +299,32 @@ $SPARK_HOME/bin/spark-shell \
        --jars ${SPARK_CUDF_JAR},${SPARK_RAPIDS_PLUGIN_JAR}
 ```  
 
+### MIG GPU on YARN
```
Comment (sameerz) [inline docs/get-started/getting-started-on-prem.md:302]: Can we add a link to some information about what MIG is, for example https://docs.nvidia.com/datacenter/tesla/mig-user-guide/index.html#introduction ?

**Comment by tgravescs** [comment]
build

**Review by jlowe — APPROVED** [review]

**Review by sameerz — APPROVED** [review]

**Comment by tgravescs** [comment]
build

**Pull request merged**